### PR TITLE
Add additional GitHub collaboration considerations (SOFTWARE-5472)

### PR DESCRIPTION
**Pull request by edquist**
TBH i feel like this Best practices section might be getting a bit long in the tooth.

But anyway, I've tried to hit all the themes mentioned in SOFTWARE-5472:

- whitespace considerations
- commit verification techniques
- PR title & body considerations
- branch name considerations
- review comment batching
- suggestion-batching considerations

I felt most of them actually belonged in the existing "Making good pull requests" subsection, though i have also added a "Reviewing pull requests" subsection for the items on that side of the collaboration process.

### CODE REVIEW

File: docs/software/git-software-development.md [hunk 1]
```diff
@@ -191,6 +191,78 @@ Instead, by putting the copy or move of the original file into its own commit,
...
+    spaces.
+    If that is the case for your editor, please configure it not to strip
+    the newline at EOF.
+    (GitHub will show the diff for files with a missing newline at EOF with
+    a red circle-minus symbol with the mouseover text "No newline at end of
+    file".)
+
+-   Avoid mixing tabs and spaces.
+    With the exception of Makefiles, indentation should be done with regular
+    spaces, not tabs.
```
Comment (matyasselmeci) [inline docs/software/git-software-development.md:217]:
```suggestion
    With the exception of Makefiles and Go source code, indentation should be done with regular
    spaces, not tabs.
```

**Review by brianhlin — APPROVED** [review]
L "GREAT" TM

File: docs/software/git-software-development.md [hunk 2]
```diff
@@ -191,6 +191,78 @@ Instead, by putting the copy or move of the original file into its own commit,
...
+    Please configure your text editor accordingly.
+    Mixing tabs and spaces in indentation is problematic because different
+    editor settings can make tab stops appear at different widths.
+
+    As with trailing whitespace, it's fine to convert existing tabs to spaces
+    on lines you are modifying, but it is not necessary to fix them everywhere,
+    if that is not the purpose of your pull request.
```
Comment (matyasselmeci) [inline docs/software/git-software-development.md:224]: I think if you're modifying a program that someone else has written, you should maintain consistency with the conventions they're using, even if you find them distasteful...

File: docs/software/git-software-development.md [hunk 3]
```diff
@@ -191,6 +191,78 @@ Instead, by putting the copy or move of the original file into its own commit,
...
+
+#### Verify that only the files intended are modified in each commit
+
+Sometimes you may have several files modified at once, but you only
+intend to commit a subset of the changes.
+In such cases you should be aware that `git commit -a` will include
+all the modified files in your commit.
+Likewise, if you have some untracked files in your working copy, that
+you do not intend to commit, be aware that `git add .` will introduce
+these as new files in the commit.
```
Comment (matyasselmeci) [inline docs/software/git-software-development.md:244]: I tend to do `git commit -av` to show the diffs in my editor while I'm writing the commit message... I find it easier to spot if I've included something I shouldn't have.

File: docs/software/git-software-development.md [hunk 4]
```diff
@@ -191,6 +191,78 @@ Instead, by putting the copy or move of the original file into its own commit,
...
+After making a commit locally, you can verify that only the files you
+intended to modify were included in the commit by running `git show --stat` .
+Or to review the actual changes to those files, `git show` (without `--stat`).
+
+If you find unintended files included in the commit, you can amend the commit
+so that it does not include changes to `file-not-to-commit` like so:
+
+            :::console
+            [user@client ~ ] $ git reset HEAD^ file-not-to-commit
+            [user@client ~ ] $ git amend
```
Comment (matyasselmeci) [inline docs/software/git-software-development.md:255]:
```suggestion
            [user@client ~ ] $ git commit --amend
```
"amend" is not a git command but I'm not surprised you have an alias for it too :-P

**Review by matyasselmeci — COMMENTED** [review]
I have a few comments...

File: docs/software/git-software-development.md (hunk 4 above)
Comment (edquist) [reply docs/software/git-software-development.md:255]: Oh heh good catch you are right :)

File: docs/software/git-software-development.md (hunk 2 above)
Comment (edquist) [reply docs/software/git-software-development.md:224]: I agree completely.  Here I had in mind fixing "obviously accidental" mixed tabs and spaces; `<SP><SP><TAB><SP><SP><SP><SP>` or similar.  Yeah i don't mean go against conventions in someone else's work; i just mean clean up stray whitespace noise.  Maybe i can make this clearer...

How does this read?

```suggestion
    As with trailing whitespace, it's fine to convert stray tabs to spaces
    on lines you are modifying, but it is not necessary to fix them everywhere,
    if that is not the purpose of your pull request.
```

File: docs/software/git-software-development.md (hunk 1 above)
Comment (edquist) [reply docs/software/git-software-development.md]: Updated.

File: docs/software/git-software-development.md (hunk 3 above)
Comment (edquist) [reply docs/software/git-software-development.md:244]: @matyasselmeci how about this?

```suggestion

    !!! note
        If you do use the `-a` option for `git commit`, you may want to
        consider using the `-v` option along with it (i.e. `git commit -av`),
        which will show you the diff to be committed in your editor while you
        are typing your commit message.

these as new files in the commit.
```

**Comment by edquist** [comment]
Alright, i think that covers @matyasselmeci's comments.

As for the CI failure, all i saw was an unrelated warning

```
WARNING -  A relative path to 'meetings/2023/TechArea202300207.md' is included in the 'nav' configuration, which is not found in the documentation files 
```

@brianhlin feel free to merge with your admin privs

**Comment by edquist** [comment]
Also, should go without saying, but feel free to hack up this dock however you guys like if you want to change it more.

**Comment by brianhlin** [comment]
Oops, thanks for the heads up on the warning. Merging

**Pull request merged**